Homework2/Task01/task01.py

```diff
@@ -0,0 +1,65 @@
+"""
+Given a file containing text. Complete using only default collections:
+    1) Find 10 longest words consisting from largest amount of unique symbols
+    2) Find rarest symbol for document
+    3) Count every punctuation char
+    4) Count every non ascii char
+    5) Find most common non ascii char for document
+"""
+import re
+import string
+from typing import List
+
+
+def open_file(file_path: str) -> str:
+    f = open(file_path)
+    return f.read()
+
+
+def get_longest_diverse_words(file_path: str) -> List[str]:
+    text = open_file(file_path)
+    word_list = re.sub(r"[^\w\s]", "", text).split()
+    all_words = [(elem, len(elem), len(set(elem))) for elem in set(word_list)]
+    sorted_words = sorted(all_words, key=lambda point: (-point[2], -point[1]))
+    return [a[0] for a in sorted_words[:10]]
+
+
+def get_rarest_char(file_path: str) -> str:
+    text = open_file((file_path))
+    counter = {}
+    for elem in text:
+        if elem not in counter:
+            counter[elem] = 1
+        else:
+            counter[elem] += 1
+    minimum = min(counter.values())
+    for num, count in counter.items():
+        if count == minimum:
+            minimum = num
```

Comment:
Should we stop here?

Reply:
I don't know how to make it in a different way. If I return num in the loop, I should return something at the end of the function. The loop already stops when we assign some value to minimum, because after 

> minimum = num

minimum has string type and we can't compare it with count anymore.

Reply:
`If I return num in the loop, I should return something at the end`

You don't need it.    If you go throw the loop and you know that value is there, why do you need something at the end?

`The loop already stops when we assign some value to minimum`
No it does not stop. You just continue looping.  And this does not have any sence. 

https://lmgtfy.app/?q=how+to+stop+loop+in+python   
  

> minimum = num

Python allows changing varibles type during execution, but it is considered a bad practice.

Reply:
> You don't need it. If you go throw the loop and you know that value is there, why do you need something at the end?

Because of that:

`"R503 you should add explicit return at end of the function if function have return value except None".`

I can ignore it or add brake and a new variable to store the answer. What is better?

Reply:
It is just a linter warning.  It is ok to ignore if you feel that you are right.

Reply:
Have already done that.